message NormL2Parameter {
  optional float spare = 1 [default = 0];
}

message ArcfaceParameter {
  optional float m = 1 [default = 0.5];
  optional bool transform_test = 2 [default = false];
}

message NoiseTolerantFRParameter {
  optional bool shield_forward = 1 [default = false];
  optional int32 start_iter = 2 [default = 1];
  optional int32 bins = 3 [default = 200];
  optional int32 slide_batch_num = 4 [default = 1000];
  optional float value_low = 5 [default = -1.0];
  optional float value_high = 6 [default = 1.0];
  optional bool debug = 7 [default = false];
  optional string debug_prefix = 8; 
}
